torch/_higher_order_ops/flex_attention.py

```diff
@@ -180,6 +186,12 @@ def math_attention(
         score_mod: The score_mod function
         other_buffers: Other buffers that are passed to the score_mod function
     """
+    query = query.view(
+        query.size(0), key.size(1), -1, query.size(-2), query.size(-1)
+    )  # [B, Hq, Mq, D] -> [B, Hkv, G, Mq, D], G=Hq//Hkv
+    key = torch.unsqueeze(key, 2)  # [B, Hkv, Mkv, D] -> [B, Hkv, 1, Mkv, D]
+    value = torch.unsqueeze(value, 2)  # [B, Hkv, Mkv, D] -> [B, Hkv, 1, Mkv, D]
+
```

Comment:
or I guess just adding the repeat interleave here

Reply:
I'll add repeat interleave here. Does repeat_interleave actually clones the tensor, or just broadcasts it?

Reply:
clones the tensor